yt/frontends/ramses/data_structures.py

```diff
@@ -18,14 +19,100 @@
 from yt.utilities.on_demand_imports import _f90nml as f90nml
 from yt.utilities.physical_constants import kb, mp
 
-from .definitions import field_aliases, particle_families, ramses_header
+from .definitions import (
+    OUTPUT_DIR_RE,
+    STANDARD_FILE_RE,
+    field_aliases,
+    particle_families,
+    ramses_header,
+)
 from .field_handlers import get_field_handlers
 from .fields import _X, RAMSESFieldInfo
 from .hilbert import get_cpu_list
 from .io_utils import fill_hydro, read_amr
 from .particle_handlers import get_particle_handlers
 
 
+class RAMSESFileSanitizer:
+    """A class to handle the different files that can be passed and associated
+    safely to a RAMSES output."""
+
+    root_folder = None  # Path | None: path to the root folder
+    info_fname = None  # Path | None: path to the info file
+    group_name = None  # str | None: name of the first group folder (if any)
+
+    def __init__(self, filename):
+        # Resolve so that it works with symlinks
+        filename = Path(filename).resolve()
+
+        self.original_filename = filename
+
+        self.output_dir = None
+        self.info_fname = None
+
+        for check_fun in (self.test_with_standard_file, self.test_with_folder_name):
+            ok, output_dir, info_fname = check_fun(filename)
+            if ok:
+                break
+
+        # Early exit if the ok flag is False
+        if not ok:
```

Comment:
```suggestion
        else:
```

Reply:
The `else` clause on a for loop is kinda confusing for folks, so I think this is `ok` since it's more explicit.

Reply:
It makes me sad that you're right.